File: _2825_make_string_a_subsequence_using_cyclic_increments/main.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        if (str2.size() > str1.size()) {
            return false;
        }

        int j = 0;
        for (int i = 0; i < str1.size() && j < str2.size(); i++) {
            if (str2[j] == str1[i] || (str2[j] == (str1[i] - 'a' + 1) % 26 + 'a')) {
                j++;
            } 
        }

        return j == str2.size();
    }
};

File: _2825_make_string_a_subsequence_using_cyclic_increments/main.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        if len(str2) > len(str1):
            return False

        i = 0
        j = 0
        while i < len(str1) and j < len(str2):
            letter1 = (ord(str1[i]) - ord('a') + 1) % 26
            letter2 = ord(str2[j]) - ord('a')

            if str1[i] == str2[j] or letter1 == letter2:
                j += 1

            i += 1

        return j == len(str2)
